protocol/communities/manager.go

```diff
@@ -841,10 +841,43 @@ func (m *Manager) markRequestToJoin(pk *ecdsa.PublicKey, community *Community) e
 	return nil
 }
 
+func (m *Manager) markRequestToJoinAsCanceled(pk *ecdsa.PublicKey, community *Community) error {
+	return m.persistence.SetRequestToJoinState(common.PubkeyToHex(pk), community.ID(), RequestToJoinStateCanceled)
+}
+
 func (m *Manager) SetMuted(id types.HexBytes, muted bool) error {
 	return m.persistence.SetMuted(id, muted)
 }
 
+func (m *Manager) CancelRequestToJoin(request *requests.CancelRequestToJoinCommunity) (*RequestToJoin, *Community, error) {
+	dbRequest, err := m.persistence.GetRequestToJoin(request.ID)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	community, err := m.GetByID(dbRequest.CommunityID)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	pk, err := common.HexToPubkey(dbRequest.PublicKey)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if err := m.markRequestToJoinAsCanceled(pk, community); err != nil {
+		return nil, nil, err
+	}
+
+	err = m.persistence.SaveCommunity(community)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	m.publish(&Subscription{Community: community})
```

Comment:
I don't think it should be here, requests to join are not published, besides that only community owners should publish community descriptions. The one who calls `CancelRequestToJoin` can't be a community owner, right?

Reply:
Actually, this is a very good catch

Reply:
fixed :) thank you :)